src/Raven.Server/Utils/RavenWindowsServiceController.cs

```diff
@@ -0,0 +1,225 @@
+                    startImmediately: true,
+                    errorSeverity: ErrorSeverity.Normal);
+
+                Console.WriteLine($@"Successfully registered service {ServiceFullName}");
+            }
+            catch (Exception e) 
+                when (e.Message.Contains("already exists"))
+            {
+                Console.WriteLine($@"Service {ServiceFullName} was already installed. Reinstalling...");
+
+                Reinstall(serviceController, serviceArgs);
+
+            } catch (Exception e) 
+                when (e.Message.Contains("The specified service has been marked for deletion"))
+            {
+                if (counter < 10)
+                {
+                    Thread.Sleep(500);
+                    counter++;
+
+                    Console.WriteLine("The specified service has been marked for deletion. Retrying {0} time", counter);
+                    InstallInternal(serviceController, serviceArgs, counter);
+                }
+            }
+
+        }
+
+        private static string NormalizeServiceName(string serviceName)
+        {
+            return Regex.Replace(serviceName, @"[\/\s]", "_");
+        }
+
+        public static void Uninstall()
+        {
+            using (var serviceController = GetServiceController())
+            {
+                UninstallInternal(serviceController);
+            }
+        }
+
+        private static void UninstallInternal(ServiceController serviceController)
+        {
+            try
+            {
+                if ((serviceController.Status == ServiceControllerStatus.Stopped || serviceController.Status == ServiceControllerStatus.StopPending) == false)
+                {
+                    StopInternal(serviceController);
+                }
+
+                new Win32ServiceManager().DeleteService(
+                    NormalizeServiceName(RavenWindowsService.WindowsServiceName));
+
+                Console.WriteLine($@"Successfully unregistered service {ServiceFullName}");
+            }
+            catch (Exception e) when (e.Message.Contains("does not exist"))
+            {
+                Console.WriteLine($@"Service {ServiceFullName} does not exist. No action taken.");
+            }
+        }
+
+        private static void Reinstall(ServiceController sc, List<string> serviceArgs)
+        {
+            StopInternal(sc);
+            UninstallInternal(sc);
+            InstallInternal(sc, serviceArgs);
+        }
+
+        private static void StopInternal(ServiceController serviceController)
+        {
+            if (!(serviceController.Status == ServiceControllerStatus.Stopped | serviceController.Status == ServiceControllerStatus.StopPending))
+            {
+                serviceController.Stop();
+                serviceController.WaitForStatus(ServiceControllerStatus.Stopped, TimeSpan.FromMilliseconds(1000));
+                Console.WriteLine($@"Successfully stopped service {ServiceFullName}");
```

Comment:
Don't we need to check if we actually stopped, we can take more than 1 sec to shut down.